### PR TITLE
Pull in eslint-config-defaults

### PR DESCRIPTION
**Pull request by baer**
This PR does 2 things:
1. Pulls in `eslint-config-defaults` without making any changes to the eslint config
2. separates test config from src config

At some point (possibly even in this PR) you may want to try turning on some of the new ES6 rules. Over time I imagine that the default that is extended (which resolves to ESLint's default values) will be overridden completely.

### CODE REVIEW

File: .eslintrc [hunk 1]
```diff
@@ -1,39 +1,9 @@
...
-globals:
-
-    ###########################################################################
-    #                                                                         #
-    #   GLOBALS: ESLint will assume the following variables are declared      #
-    #   globally; other variables will require explicit declaration.          #
-    #                                                                         #
-    ###########################################################################
-
-    require: true
```
Comment (baer) [inline .eslintrc]: Obviated when you use env mocha. ESlint uses the globals package below.

https://github.com/sindresorhus/globals/blob/master/globals.json#L683-L703

File: .eslintrc [hunk 2]
```diff
@@ -1,39 +1,9 @@
...
-    beforeEach: true
-    afterEach: true
-    before: true
-    after: true
-    it: true
-    sinon: true
-    expect: true
+extends:
+  - "defaults"
+  - "defaults/rules/eslint/node/walmart"
```
Comment (baer) [inline .eslintrc]: You were pretty close to the WM config so I aligned that section to override it but I can pull in the "off.js" file instead and let you 100% override or whatever you want to do. I just wanted to use this PR to get the ball rolling.

**Comment by divmain** [comment]
Definitely want to use eslint-config-defaults.  However, my preference is to have a single `.eslintrc` file.  I expect you may disagree with me on this :) and I would understand why.  Having multiple is more correct.  However, it makes integration with my editor more difficult, and working with my editor comprises 95% of the feedback loop for me in fixing lint errors.

I suppose I wouldn't mind having a _more_ strict version for tests or app code (although it makes the dir feel cluttered to me).  But most importantly, I want an almost perfect `.eslintrc` file that works for everything.

**Comment by baer** [comment]
Fixed. Editor integration is a pretty sound reason to move to a single config. I don't have string opinions here so lets just get this in and start honing down / turning on rules :)

**Comment by divmain** [comment]
Excellent :+1:

**Pull request merged**